lib/df-detection.c

```diff
@@ -67,7 +67,8 @@ void report(void)
 	pr_err("BUG: Intersection Detected \n ");
 	pr_err("==================================================================\n");
 	if (panic_on_warn) {
```

Comment:
We actually need to always print the report (regardless of whether we want panic_on_warn or not), but then call panic() if panic_on_warn is set.

Reply:
but when i tried to print report without panic_on_warn non of the prints appear on syzkaller report , only when i used panic_on_warn all the data we wanted appears in syzkaller report